onnxruntime/core/providers/nnapi/nnapi_builtin/builders/op_builder.cc

```diff
@@ -1608,9 +1607,24 @@ bool GemmOpBuilder::IsQuantizedOp(const NodeUnit& node_unit) const {
 void GemmOpBuilder::AddInitializersToSkip(ModelBuilder& model_builder, const NodeUnit& node_unit) const {
   const auto& inputs = node_unit.Inputs();
   if (IsQuantizedOp(node_unit)) {
-    AddQuantizationScaleAndZeroPointToSkip(model_builder, *inputs[0].quant_param);               // b_scale, b_zp
-    AddInputToSkip(model_builder, inputs[1]);                                                    // b, b_scale, b_zp
-    AddQuantizationScaleAndZeroPointToSkip(model_builder, *node_unit.Outputs()[0].quant_param);  // y_scale, y_zp
+    if (node_unit.OpType() == "QLinearMatMul" || node_unit.OpType() == "MatMul") {                 // QLinearMatMul/QDQMatMul
+      AddQuantizationScaleAndZeroPointToSkip(model_builder, *inputs[0].quant_param);               // a_scale, a_zp
+      AddInputToSkip(model_builder, inputs[1]);                                                    // b, b_scale, b_zp
+      AddQuantizationScaleAndZeroPointToSkip(model_builder, *node_unit.Outputs()[0].quant_param);  // y_scale, y_zp
+    } else if (node_unit.OpType() == "Gemm") {                                                     // QDQGemm
+      AddQuantizationScaleAndZeroPointToSkip(model_builder, *inputs[0].quant_param);               // x_scale, x_zp
+      AddQuantizationScaleAndZeroPointToSkip(model_builder, *inputs[1].quant_param);               // w_scale, w_zp
+
```

Comment:
a_scale/a_zp and b_scale/b_zp?

Reply:
sorry I realized the set of comments " x_scale/x_zp, w_scale/w_zp, B_scale/B_zp" causes a bit confusion. actually it comes from the other comments I was looking at from other qdq operator support.  The `B` stands for bias not the input B of gemm operator. sorry for the confusion. I can switch all of them back to align with `A*B+C`